core/cloudflow-operator/src/main/scala/cloudflow/operator/action/Action.scala

```diff
@@ -102,15 +118,22 @@ object Action {
   /**
    * Log message for when an [[Action]] is about to get executed.
    */
-  def executing(action: Action[ObjectResource]) =
-    s"Executing ${action.name} action for resource ${action.resource.kind}/${action.resource.metadata.name} in namespace ${action.resource.namespace}"
+  def executing(action: Action[ObjectResource]) = action.executing
 
   /**
    * Log message for when an [[Action]] has been executed.
    */
-  def executed(action: Action[ObjectResource]) =
-    s"Executed ${action.name} action for ${action.resource.kind} ${action.resource.metadata.name} in namespace ${action.resource.namespace}"
+  def executed(action: Action[ObjectResource]) = action.executed
+}
 
+abstract class ResourceAction[T <: ObjectResource] extends Action[T] {
+  def resource: T
+  def resourceName = resource.metadata.name
+  def namespace    = resource.metadata.namespace
+  def executing =
```

Comment:
I like the idea of adding some state markers. Yet, I find that this approach could be better structured.
The difference in the messages between `executing` and `executed` is very small every time. It also follows a pattern we could exploit:
`<action> for <identifier>` 
Not really a blocker right now.

Reply:
yeah this is just refactoring what we had already.. The delete action was a bit weird that it needed the total resource to delete it, so I modified that.